aopy/data/bmi3d.py

```diff
@@ -564,4 +566,127 @@ def get_source_files(preproc_dir, subject, te_id, date):
+
+        # Trial aligned event codes and event times
+        tr_seg, tr_t = get_trial_segments_and_times(event_codes, event_times, trial_start_codes, trial_end_codes)
+
+        # Get data segments 
+        cursor_traj = get_kinematic_segments(preproc_dir, subject, te, date, trial_start_codes, trial_end_codes, 
+                                                         datatype='cursor')[0].tolist()
+        hand_traj = [None] * len(tr_seg)
+        if include_handdata:
+            hand_traj = get_kinematic_segments(preproc_dir, subject, te, date, trial_start_codes, trial_end_codes, 
+                                                  datatype='hand')[0].tolist()
+        eye_traj = [None] * len(tr_seg)
+        if include_eyedata:
+            eye_traj = get_kinematic_segments(preproc_dir, subject, te, date, trial_start_codes, trial_end_codes, 
+                                                 datatype='eye')[0].tolist()
+
+        target_idx = [code[np.isin(code, target_codes)][0] - target_codes[0] if np.sum(np.isin(code, target_codes)) == 1 else 0 for code in tr_seg]
+        target_location = get_target_locations(preproc_dir, subject, te, date, target_idx).tolist()
+
+        reward = [np.any(np.isin(reward_codes, ec)) for ec in tr_seg]
+        penalty = [np.any(np.isin(penalty_codes, ec)) for ec in tr_seg]
+        
+        df = pd.concat([df,pd.DataFrame({'subject': subject,
+                                         'te_id': te, 
+                                         'date': date, 
+                                         'event_codes': tr_seg,
+                                         'event_times': tr_t, 
+                                         'reward': reward,
+                                         'penalty': penalty,
+                                         'target_idx': target_idx,
+                                         'target_location': target_location,
+                                         'cursor': cursor_traj, 
+                                         'hand': hand_traj, 
+                                         'eye': eye_traj,
+                                        })], ignore_index=True)
+    
+    return df
+
+def concat_trials_center_out(preproc_dir, subjects, ids, dates, df=None, include_center_target=True,
+                  include_handdata=False, include_eyedata=False):
+    '''
+    Wrapper around concat_trials() specifically for center-out experiments.
+
+    Args:
+        preproc_dir (str): base directory where the files live
+        subjects (list of str): Subject name for each recording
+        ids (list of int): Block number of Task entry object for each recording
+        dates (list of str): Date for each recording
+        df (DataFrame, optional): pandas DataFrame object to append. Defaults to None.
+        include_center_target (bool, optional): If True, trials begin after the cursor enters
+            the center target. Otherwise trials begin after the go cue. Default True.
```

Comment:
Is there a way to also allow the user to pick when the cursor is leaves the center target?

Reply:
Leave center target is roughly equivalent to center target turning off. If you wanted cursor leaving the center target radius, you could start with cursor enter center target and then later figure out when the trajectory leaves that radius.